magento/magento2
# Refresh invalidated caches nightly

### Description
When certain actions are taken in the admin, related caches are marked as invalid. Manual action is required to refresh these caches. This is often okay as the action which invalidated the cache(s) was performed by an admin user. However there are some automated actions which can invalidate caches. In this case, caches may remain in an invalidated state for long periods of time, resulting in hard-to-diagnose (unexpected) behaviour.

This pull request introduces a cronjob which will refresh all invalidated caches.

### Manual testing scenarios
1. Make a configuration change in the admin
1. Observe that the Configuration and Full Page caches are marked as invalidated
1. Run the (existing) `backend_clean_cache` cronjob
1. Observe that the Configuration and Full Page caches are still marked as invalidated
1. Run the `backend_refresh_invalidated_caches` crobjob (which is introduced in this pull request)
1. Observe that no caches are marked as invalidated

### Contribution checklist


### Resolved issues:
1. [x] resolves magento/magento2#40337: Refresh invalidated caches nightly
